core/src/main/java/hudson/cli/DisablePluginCommand.java

```diff
@@ -68,68 +78,161 @@ protected int run() throws Exception {
         Jenkins jenkins = Jenkins.get();
         jenkins.checkPermission(Jenkins.ADMINISTER);
 
-        PluginManager manager = jenkins.getPluginManager();
-        int result = 0;
-        boolean somePluginDisabled = false; // to ensure we restart when it needs to
-        for (String pluginName : pluginNames) {
-            switch (disablePlugin(manager, pluginName)) {
-                case DISABLED:
-                    if(!somePluginDisabled) somePluginDisabled = true;
-                    break;
-                case NOT_DISABLED_DEPENDANTS:
-                    if(result != 16) result = 16; // Custom error code to indicate one plugin, at least, can't be disabled
+        // get the strategy as an enum
+        PluginWrapper.PluginDisableStrategy strategyToUse;
+        try {
+            strategyToUse = PluginWrapper.PluginDisableStrategy.valueOf(strategy.toUpperCase());
+        } catch (IllegalArgumentException iae) {
+            throw new IllegalArgumentException(hudson.cli.Messages.DisablePluginCommand_NoSuchStrategy(strategy, String.format("%s, %s, %s", PluginWrapper.PluginDisableStrategy.NONE, PluginWrapper.PluginDisableStrategy.MANDATORY, PluginWrapper.PluginDisableStrategy.ALL)));
+        }
+
+        // disable...
+        List<PluginWrapper.PluginDisableResult> results = jenkins.pluginManager.disablePlugins(strategyToUse, pluginNames);
+
+        // print results ...
+        printResults(results);
+
+        // restart if it was required and it's necessary (some plugin was disabled in this execution)
```

Comment:
some plugin was disabled in this execution => some plugin**s** w**ere** disabled in this execution

Reply:
It's intentionally singular because you just need one plugin disabled in this execution to raise the restart, I will change it by _at least one plugin was..._

Reply:
In that case that's perfectly fine 👍